Core/include/Acts/TrackFitting/GainMatrixUpdater.hpp

```diff
@@ -61,39 +49,42 @@ class GainMatrixUpdater {
 
     // read-only handles. Types are eigen maps to backing storage
     const auto predicted = trackState.predicted();
-    const auto predicted_covariance = trackState.predictedCovariance();
+    const auto predictedCovariance = trackState.predictedCovariance();
 
     ACTS_VERBOSE("Predicted parameters: " << predicted.transpose());
-    ACTS_VERBOSE("Predicted covariance:\n" << predicted_covariance);
+    ACTS_VERBOSE("Predicted covariance:\n" << predictedCovariance);
 
     // read-write handles. Types are eigen maps into backing storage.
     // This writes directly into the trajectory storage
     auto filtered = trackState.filtered();
-    auto filtered_covariance = trackState.filteredCovariance();
+    auto filteredCovariance = trackState.filteredCovariance();
 
-    std::optional<std::error_code> error{std::nullopt};  // assume ok
+    // default error represents an invalid non-error state, no need for optional
+    std::error_code error;
     visit_measurement(
         trackState.calibrated(), trackState.calibratedCovariance(),
         trackState.calibratedSize(),
-        [&](const auto calibrated, const auto calibrated_covariance) {
-          constexpr size_t measdim = decltype(calibrated)::RowsAtCompileTime;
-          using cov_t = ActsSymMatrixD<measdim>;
-          using par_t = ActsVectorD<measdim>;
-
-          ACTS_VERBOSE("Measurement dimension: " << measdim);
+        [&](const auto calibrated, const auto calibratedCovariance) {
+          constexpr size_t kMeasurementSize =
+              decltype(calibrated)::RowsAtCompileTime;
+          using Scalar = typename decltype(calibrated)::Scalar;
+          using ParametersVector = ActsVector<Scalar, kMeasurementSize>;
+          using CovarianceMatrix = ActsSymMatrix<Scalar, kMeasurementSize>;
+
+          ACTS_VERBOSE("Measurement dimension: " << kMeasurementSize);
           ACTS_VERBOSE("Calibrated measurement: " << calibrated.transpose());
           ACTS_VERBOSE("Calibrated measurement covariance:\n"
-                       << calibrated_covariance);
+                       << calibratedCovariance);
 
-          const ActsMatrixD<measdim, eBoundSize> H =
+          const auto& H =
```

Comment:
Since this is a copy, I'd prefer to not rely on lifetime extension, and make it explicit.

Reply:
I switched back to storing the result by value using a concrete type. Apparently this change also broke the unit tests.